EventFilter/Phase2PixelRawToDigi/plugins/PixelToRawProducer.cc

```diff
@@ -0,0 +1,257 @@
+// ED producer that creates FEDRawDataCollection from inner tracker quarer cores
+#include <utility>
+#include <unordered_map>
+#include <string>
+#include <iostream>
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/one/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "Geometry/CommonDetUnit/interface/PixelGeomDetUnit.h"
+
+#include "CondFormats/SiPhase2TrackerObjects/interface/TrackerDetToDTCELinkCablingMap.h"
+#include "CondFormats/SiPhase2TrackerObjects/interface/DTCELinkId.h"
+#include "CondFormats/DataRecord/interface/TrackerDetToDTCELinkCablingMapRcd.h"
+
+#include "DataFormats/Phase2TrackerDigi/interface/Phase2ITChipBitStream.h"
+#include "DataFormats/Common/interface/DetSetVector.h"
+#include "DataFormats/FEDRawData/interface/FEDRawDataCollection.h"
+#include "DataFormats/FEDRawData/interface/FEDRawData.h"
+#include "DataFormats/FEDRawData/interface/FEDHeader.h"
+#include "DataFormats/FEDRawData/interface/FEDTrailer.h"
+
+class PixelToRawProducer : public edm::one::EDProducer<> {
+public:
+    explicit PixelToRawProducer(const edm::ParameterSet&);
+    ~PixelToRawProducer() override;
+
+private:
+    void produce(edm::Event&, const edm::EventSetup&) override;
+    
+    const edm::ESGetToken<TrackerDetToDTCELinkCablingMap, TrackerDetToDTCELinkCablingMapRcd> cablingMapToken_;
+    const edm::EDGetTokenT<edm::DetSetVector<Phase2ITChipBitStream>> ITChipBitStreamToken_;
+
+    unsigned int IndexingDTC(unsigned int dtc_id);
+    void AddHexToPtr(unsigned char *data_ptr, int globalIndex, int localIndex, const std::vector<bool>& moduleBitStream);
+    void PrintBitVectorAs16bit(const std::vector<bool>& bits, const std::string& label);
+
+    static constexpr int SLINKS_PER_DTC = 16;
+    static constexpr int MIN_DTC_ID = 11;
```

Comment:
Surely the DTC IDs should come from the cabling map? Why do you need to hard-wired the minimum and maximum one? Did the OT unpacker do this?

Reply:
ah my bad this was just hard coded thing i did for the time being. i'll fix this

Reply:
made use of cabling map (added some more functionalities in condformats as well to make it happen)